### ISSUE TITLE
Reconnect does not restore client subscriptions

### ISSUE THREAD

**Issue by andig** [philippseith/signalr#110]
I'm creating the client like this:

	client, err := signalr.NewClient(context.Background(),
		signalr.WithAutoReconnect(c.connect(ts)),
		signalr.WithReceiver(c),
		signalr.Logger(easee.SignalrLogger(c.log.TRACE), true),
	)

When I connect to the remote server and subscribe for updates I'm greeted with a bunch of messages:

    [easee ] TRACE 2021/11/30 21:41:48 level=debug caller=client.go:259 state=1
    [easee ] TRACE 2021/11/30 21:41:48 POST https://api.easee.cloud/hubs/chargers/negotiate
    [easee ] TRACE 2021/11/30 21:41:48 connection=-gVGevD2xUgR5IBoUYV0jw level=debug caller=client.go:523 event=handshake sent msg={"protocol":"json","version":1}
    [easee ] TRACE 2021/11/30 21:41:48 connection=-gVGevD2xUgR5IBoUYV0jw level=debug caller=client.go:553 event=handshake received msg=signalr.handshakeResponse{Error:""}
    [easee ] TRACE 2021/11/30 21:41:48 level=debug caller=client.go:259 state=2
    [easee ] TRACE 2021/11/30 21:41:48 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":1,"target":"SubscribeWithCurrentState","invocationId":"1","arguments":["EH635870",true]}
    [easee ] TRACE 2021/11/30 21:41:48 level=debug caller=jsonhubprotocol.go:76 event=read message={"type":1,"target":"ProductUpdate","arguments":[{"mid":"EH635870","dataType":2,"id":15,"timestamp":"2021-11-30T12:10:01Z","value":"1"}]}

However, after reconnect (forced by disabling Wifi), data is no longer received:

    [easee ] TRACE 2021/11/30 21:44:47 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:44:52 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:44:57 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:02 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:07 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:12 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:13 level=debug caller=jsonhubprotocol.go:76 event=read message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:13 level=debug caller=jsonhubprotocol.go:76 event=read message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:13 connection=at72LoSuEwXQd7qK42Yoog level=debug caller=loop.go:285 event=message received message=signalr.hubMessage{Type:6}
    [easee ] TRACE 2021/11/30 21:45:13 connection=at72LoSuEwXQd7qK42Yoog level=debug caller=loop.go:285 event=message received message=signalr.hubMessage{Type:6}
    [easee ] TRACE 2021/11/30 21:45:13 level=debug caller=jsonhubprotocol.go:76 event=read message={"type":7,"allowReconnect":true}
    [easee ] TRACE 2021/11/30 21:45:13 connection=at72LoSuEwXQd7qK42Yoog level=debug caller=loop.go:91 {{7  true} <nil>}=message received message=signalr.closeMessage{Type:7, Error:"", AllowReconnect:true}
    [easee ] TRACE 2021/11/30 21:45:13 connection=at72LoSuEwXQd7qK42Yoog level=debug caller=loop.go:130 event=message loop ended
    [easee ] TRACE 2021/11/30 21:45:13 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":7,"error":"","allowReconnect":false}
    [easee ] TRACE 2021/11/30 21:45:13 level=info connect=*signalr.webSocketConnection: failed to write msg: WebSocket closed: received close frame: status = StatusNormalClosure and reason = ""
    [easee ] TRACE 2021/11/30 21:45:14 level=debug caller=client.go:259 state=1
    [easee ] TRACE 2021/11/30 21:45:14 POST https://api.easee.cloud/hubs/chargers/negotiate
    [easee ] TRACE 2021/11/30 21:45:16 connection=-dB9hjF7o1oTAxjYC5pKuw level=debug caller=client.go:523 event=handshake sent msg={"protocol":"json","version":1}
    [easee ] TRACE 2021/11/30 21:45:16 connection=-dB9hjF7o1oTAxjYC5pKuw level=debug caller=client.go:553 event=handshake received msg=signalr.handshakeResponse{Error:""}
    [easee ] TRACE 2021/11/30 21:45:16 level=debug caller=client.go:259 state=2
    [easee ] TRACE 2021/11/30 21:45:21 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:26 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:31 level=debug caller=jsonhubprotocol.go:76 event=read message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:31 connection=-dB9hjF7o1oTAxjYC5pKuw level=debug caller=loop.go:285 event=message received message=signalr.hubMessage{Type:6}
    [easee ] TRACE 2021/11/30 21:45:36 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}
    [easee ] TRACE 2021/11/30 21:45:41 level=debug caller=jsonhubprotocol.go:211 event=write message={"type":6}

Is it possible that the subscriptions are not restored?

**Comment by andig** [comment]
For the sake of completeness, and may it's relevant for the readme, here's what I did to control initial connect and restore subscriptions on reconnect:

	client, err := signalr.NewClient(context.Background(),
		signalr.WithAutoReconnect(c.connect(ts)),
		signalr.WithReceiver(c),
		signalr.Logger(easee.SignalrLogger(c.log.TRACE), false),
	)

	if err == nil {
		c.subscribe(client)

		client.Start()

		// wait for initial connection with timeout
		ctx, cancel := context.WithTimeout(context.Background(), request.Timeout)
		defer cancel()
		err = <-client.WaitForState(ctx, signalr.ClientConnected)
	}

	// wait for first update received after subscription
	c.mux.L.Lock()
	c.mux.Wait()
	for c.updated.IsZero() {
		c.mux.Wait()
	}
	c.mux.L.Unlock()

with `subscribe`:

	stateC := make(chan struct{}, 1)
	_ = client.ObserveStateChanged(stateC)

	go func() {
		for range stateC {
			if client.State() == signalr.ClientConnected {
				if err := <-client.Send("SubscribeWithCurrentState", c.charger, true); err != nil {
					c.log.ERROR.Printf("SubscribeWithCurrentState: %v", err)
				}
			}
		}
	}()

and `connect`:

	return func() (signalr.Connection, error) {
		tok, err := ts.Token()
		if err != nil {
			return nil, err
		}

		ctx, cancel := context.WithTimeout(context.Background(), request.Timeout)
		defer cancel()

		return signalr.NewHTTPConnection(ctx, "https://api.easee.cloud/hubs/chargers",
			signalr.WithHTTPClient(c.Client),
			signalr.WithHTTPHeaders(func() (res http.Header) {
				return http.Header{
					"Authorization": []string{fmt.Sprintf("Bearer %s", tok.AccessToken)},
				}
			}),
		)
	}

with receiving data:

    // handle data, then confirm received
    c.mux.Lock()
    c.updated = time.Now()
    c.mux.Unlock()
    
    // unblock goroutine
	c.mux.Broadcast()

**Comment by philippseith** [comment]
It would be nice to have a sample with this code in the repo and to link it in the Readme.
